xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodec.cpp

```diff
@@ -116,8 +116,8 @@ CRenderInfo CRendererMediaCodec::GetRenderInfo()
 {
   CRenderInfo info;
   info.formats = m_formats;
-  info.max_buffer_size = NUM_BUFFERS;
-  info.optimal_buffer_size = 2;
+  info.max_buffer_size = 4;
```

Comment:
Those are hardware buffers locked by Kodi, so the lowest, the best.
3 seems an actual minimum for proper working, nowadays, and "max" doesn't seem to be actually used, but 6 is too much.